### ISSUE TITLE
check_disk reporting path as "DISK CRITICAL ... 0 MB (0% inode=0%)" for no obvious reason

### ISSUE THREAD

**Issue by christian-2** [monitoring-plugins/monitoring-plugins#1819]
I am observing a situation where `check_disk` apparently reports wrong metrics. This is on a Kubernetes node (running Debian 11) where several pods mount `ConfigMaps`, hence there are several mount points for `/dev/mapper/rootvg-var` on that node.

```
# df -hT /dev/mapper/rootvg-var
Filesystem             Type  Size  Used Avail Use% Mounted on
/dev/mapper/rootvg-var ext4   61G   44G   15G  75% /var
# sudo -u nagios /usr/lib/nagios/plugins/check_disk -c 10% -w 20% -p /dev/mapper/rootvg-var
DISK CRITICAL - free space: /var/lib/kubelet/pods/9c6f2c28-9aa2-49da-bd75-9a3ed5e4bfc7/volume-subpaths/prometheus-recording-rules/prometheus/3 0 MB (0% inode=0%);| /var/lib/kubelet/pods/9c6f2c28-9aa2-49da-bd75-9a3ed5e4bfc7/volume-subpaths/prometheus-recording-rules/prometheus/3=7635037788430MB;-2147483648;-2147483648;0;7635037788430
# echo $?
2
# mount | grep "^\/dev\/mapper\/rootvg-var"
/dev/mapper/rootvg-var on /var type ext4 (rw,relatime)
...
/dev/mapper/rootvg-var on /var/lib/kubelet/pods/9c6f2c28-9aa2-49da-bd75-9a3ed5e4bfc7/volume-subpaths/prometheus-recording-rules/prometheus/3 type ext4 (rw,relatime)
# dpkg -S /usr/lib/nagios/plugins/check_disk
monitoring-plugins-basic: /usr/lib/nagios/plugins/check_disk
# dpkg -l monitoring-plugins-basic
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name                     Version      Architecture Description
+++-========================-============-============-========================================================
ii  monitoring-plugins-basic 2.3.1-1      amd64        Plugins for nagios compatible monitoring systems (basic)
```

In the UI (Icinga Web 2) the result is rendered as CRITICAL "DISK CRITICAL - free space: /var/lib/kubelet/pods/9c6f2c28-9aa2-49da-bd75-9a3ed5e4bfc7/volume-subpaths/prometheus-recording-rules/prometheus/3 0 MB (0% inode=0%);" which is clearly incorrect, since `/dev/mapper/rootvg-var` has 15G available. What is going on here?

**Comment by RincewindsHat** [comment]
Hi @christian-2, that sounds ugly. what does `df /var/lib/kubelet/pods/9c6f2c28-9aa2-49da-bd75-9a3ed5e4bfc7/volume-subpaths/prometheus-recording-rules/prometheus/3` say?

**Comment by christian-2** [comment]
@RincewindsHat Thanks for your quick response. Here's what I am observing (so perhaps the lack of permission might be the root cause):

```
# whoami
root
# df /var/lib/kubelet/pods/9c6f2c28-9aa2-49da-bd75-9a3ed5e4bfc7/volume-subpaths/prometheus-recording-rules/prometheus/3
Filesystem             1K-blocks     Used Available Use% Mounted on
/dev/mapper/rootvg-var  63931884 45618884  15601180  75% /var/lib/kubelet/pods/9c6f2c28-9aa2-49da-bd75-9a3ed5e4bfc7/volume-subpaths/prometheus-recording-rules/prometheus/3
# sudo -u nagios df /var/lib/kubelet/pods/9c6f2c28-9aa2-49da-bd75-9a3ed5e4bfc7/volume-subpaths/prometheus-recording-rules/prometheus/3
df: /var/lib/kubelet/pods/9c6f2c28-9aa2-49da-bd75-9a3ed5e4bfc7/volume-subpaths/prometheus-recording-rules/prometheus/3: Permission denied
# sudo -u nagios df /var
Filesystem             1K-blocks     Used Available Use% Mounted on
/dev/mapper/rootvg-var  63931884 45624612  15595452  75% /var
```

**Comment by christian-2** [comment]
@RincewindsHat Apparently there is no `monitoring-plugins-basic=2.3.2-1` available for Debian 11 or from its backports yet, and I do not see any mentioning of `check_disk` in the relevant change logs.

But I do have the impression that `check_disk` already supports multiple mountpoints per device plus a notion of selecting the "best" matching one (see e.g. [here](https://github.com/monitoring-plugins/monitoring-plugins/blob/f0ac7fcc7c40fab04c00fbbc8c091e89e77b0f74/gl/mountlist.c#L439) and [here](https://github.com/monitoring-plugins/monitoring-plugins/blob/f0ac7fcc7c40fab04c00fbbc8c091e89e77b0f74/lib/utils_disk.c#L121)). Perhaps it assumes that it possesses full access permissions when it attempts the selection.

Perhaps I'll install the check from GitHub sources and see if I can confirm this hypothesis and come up with a PR, time permitting.

**Comment by RincewindsHat** [comment]
indeed, a PR would be great. Thank you for the report and the effort.

**Comment by christian-2** [comment]
@RincewindsHat FYI I have submitted PR #1820 (fixes #1819), as discussed

**Comment by RincewindsHat** [comment]
:+1:
